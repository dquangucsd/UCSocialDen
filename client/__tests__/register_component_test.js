import 'react-native';
import React from 'react';
import RegisterScreen from '../app/register';
import { render } from '@testing-library/react-native';



describe('RegisterScreen', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<RegisterScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});


