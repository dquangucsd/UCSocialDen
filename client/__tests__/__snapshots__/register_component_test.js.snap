// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#556ebe",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Zain",
          "fontSize": 34,
          "fontWeight": "bold",
        }
      }
    >
      UC Social Den
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fafafa",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#556ebe",
          "fontFamily": "Zain",
          "fontSize": 32,
          "fontWeight": "bold",
          "marginBottom": 30,
          "marginTop": 100,
        }
      }
    >
      Create an Account
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Major"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#556ebe",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginBottom": 15,
          "padding": 10,
          "width": "80%",
        }
      }
      value=""
    />
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Bio (Optional)"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#556ebe",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginBottom": 15,
          "padding": 10,
          "width": "80%",
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#4285F4",
          "borderRadius": 5,
          "marginBottom": 15,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Upload Profile Photo
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#4285F4",
          "borderRadius": 5,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
